sound/soc/sof/intel/hda-dai.c

```diff
@@ -447,6 +480,9 @@ static int ssp_dai_hw_params(struct snd_pcm_substream *substream,
 			/* we can return immediately as there are no choises */
 			if (sof_dai->number_configs < 2)
 				config = &sof_dai->dai_config[sof_dai->current_config];
+			/* check for possible enum control set by user */
+			if (!config)
+				config = ssp_dai_config_control_match(sdev, sof_dai);
```

Comment:
this seems moot now no? You'd ignore this and go ahead and match with pcm params anyway? So whats the point of the kcontrol then?

Reply:
ok, this part will change